### PR TITLE
docs(headless): add React samples for headless results-per-page

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-388

### CODE REVIEW

**Review by olamothe — APPROVED** [review]

File: packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  const {controller, options} = props;
+  const [, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  const isSetTo = (numberOfResults: number) => {
+    return controller.isSetTo(numberOfResults);
+  };
+
+  const set = (numberOfResults: number) => {
```
Comment (nrawji) [inline packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx:25]: Why do we need these wrapping methods?

**Review by nrawji — APPROVED** [review]

File: packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx (hunk 1 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx:25]: Hmm, looking back, I can't find the reason why I did this. I recalled trying to stay consistent with something else, but I can't find any other instance of wrapping methods being used like this. I'll remove them.

File: packages/samples/headless-react/src/components/results-per-page/results-per-page.class.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import {Component} from 'react';
+import {
+  buildResultsPerPage,
+  ResultsPerPage as HeadlessResultsPerPage,
+  ResultsPerPageState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+const options = [10, 25, 50, 100];
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/results-per-page/results-per-page.class.tsx:10]: Could we make this variable a property inside the class?

File: packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx (hunk 1 above)
Comment (samisayegh) [reply packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx:25]: Let's do the same with the class component please,

**Review by samisayegh — APPROVED** [review]

**Pull request merged**